src/app/shared/pipes/auth-provider-name.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable, Pipe, PipeTransform} from '@angular/core'
+import {AuthProvider} from "../../preference/state/preference.store"
+
+@Injectable({
+  providedIn: 'root',
+})
+@Pipe({
+  name: 'authProviderName',
+})
+export class AuthProviderNamePipe implements PipeTransform {
+  constructor() {
+  }
+
+  transform(value: '' | AuthProvider | null | undefined): string {
+    if (!value) {
+      return ''
+    }
+
+    switch (value as AuthProvider) {
+      case AuthProvider.METAMASK:
+        return 'Metamask'
+      case AuthProvider.WALLET_CONNECT:
+        return 'Wallet Connect'
+      case AuthProvider.VENLY:
+        return 'Venly'
+      default:
+        throw new Error("Unknown auth provider")
```

Comment:
I understand the intent, but it's not usual to throw errors in pipes because you cannot handle them. Rather return a simple text like `'Unknown'`, or `'Unknown Auth Provider'`, or just `'Provider'`... as we'll know why that happened.
Additionally, you can add a log to indicate the error in the console as: `console.error('Unknown auth provider')`.

Reply:
imho the proper and type safe way would be for the pipe to return (string | null) as the return type. That way, the consumer of the pipe will be forced to decide at point of consumption how to handle the nullable type.  

This way it just fails - silently returning a wrong value

Reply:
That's if it's possible to return a null from a pipe, not sure

Reply:
`null` can be returned from a pipe and nothing will be displayed in that case. I tried out Angular's `date` pipe with invalid string input, and it shows nothing - that means that it returns either `null` or `''`.

I just figured out that we can omit the `default` case in this pipe since this won't compile:
```typescript
transform(value: '' | AuthProvider | null | undefined): string {
    if (!value) {
      return ''
    }

    switch (value as AuthProvider) {
      case AuthProvider.METAMASK:
        return 'Metamask'
      case AuthProvider.WALLET_CONNECT:
        return 'Wallet Connect'
      // case AuthProvider.VENLY:
      //   return 'Venly'
    }
  }
```
TypeScript knows that we want to return non-nullable `string` and that we are missing a branch in switch-case, so there is a compile error.